pvlib/shading.py

```diff
@@ -553,3 +553,109 @@ def shaded_fraction1d(
     )
 
     return np.clip(t_asterisk, 0, 1)
+
+
+def martinez_shade_factor(shaded_fraction, N_shaded_blocks, N_total_blocks):
+    r"""
+    A shading correction factor for the direct and circumsolar incident
+    irradiance of non-monolithic Silicon
+    modules and arrays with an arbitrary number of bypass diodes.
+
+    .. versionadded:: 0.11.0
+
+    Parameters
+    ----------
+    shaded_fraction : numeric
+        Surface shaded fraction. Unitless.
+    shaded_blocks : numeric
+        Number of blocks affected by the shadow. Unitless integer.
+        If a floating point number is provided, it will be rounded up.
+    total_blocks : numeric
+        Number of total blocks. Unitless integer.
+
+    Returns
+    -------
+    shading_correction_factor : numeric
+        Multiply direct and circumsolar irradiance by this factor.
+
+    Notes
+    -----
+    The implemented equation is (6) from [1]_:
+
+    .. math::
+
+        (1 - F_{ES}) = (1 - F_{GS}) (1 - \frac{N_{SB}}{N_{TB} + 1})
+
+    Where :math:`(1 - F_{ES})` is the correction factor to be multiplied by
+    the direct and circumsolar irradiance, :math:`F_{GS}` is the shaded
+    fraction of the collector, :math:`N_{SB}` is the number of shaded blocks
+    and :math:`N_{TB}` is the number of total blocks.
+
+    Blocks terminology
+    ^^^^^^^^^^^^^^^^^^
+    [1]_ defines a *block* as a group of solar cells protected by a bypass
+    diode. Also, a *block* is shaded when at least one of its cells is shaded.
+
+    How many blocks and their layout depend on the module(s) used. Many
+    manufacturers don't specify this information explicitly. However, we can
+    infer these values from:
+
+    - the number of bypass diodes
+    - where and how many junction boxes are present on the back of the module
+    - whether or not the module is comprised of *half-cut cells*
+
+    The latter two are heavily correlated.
+
+    For example:
+
+    1. A module with 1 bypass diode behaves as 1 block.
+    2. A module with 3 bypass diodes and 1 junction box is likely to have 3
+       blocks.
+    3. A half-cut module with 3 junction boxes (split junction boxes) is
+       likely to have 3x2 blocks. The number of blocks along the longest
+       side of the module is 2 and along the shortest side is 3.
+    4. A module without bypass diodes doesn't constitute a block, but may be
+       part of one.
+
+    .. figure:: _images/Centralized_and_split_PV_junction_boxes_cesardd.jpg
+       :alt: Centralized and split PV junction boxes.
```

Comment:
This image isn't rendering on RTD

Reply:
Removing it too as in https://github.com/pvlib/pvlib-python/pull/2070#discussion_r1623525705, thou IDK if that's a good choice. I don't have better images to describe the blocks.